### PR TITLE
feat(deployment/daemonset): Remove imagePullPolicy from charts

### PR DESCRIPTION
**Pull request by Mart-Kuc**
#### Pull Request Checklist ####

- [ ] Any new images or tags consumed by charts has been added [here](https://github.com/rancher/image-mirror)
- [ ] Chart version has been incremented (if necessary)
- [x] That helm lint and pack run successfully on the chart.
- [x] Deployment of the chart has been tested and verified that it functions as expected.
- [ ] Changes to scripting or CI config have been tested to the best of your ability

#### Types of Change ####
- Remove `ImagePullPolicy: "Always"` from `daemonset` and `deployment`

According to [documentation](https://kubernetes.io/docs/concepts/containers/images/#imagepullpolicy-defaulting) if you specify version of image tag, automatically ImagePullPolicy is set to `IfNotPressent`.
And vice versa if you not specify version of image tag (keep latest) ImagePullPolicy is set to `Always`.

So keep ImagePullPolicy hard coded to `Always` in charts does not make sense.
It overrides to `Always` even if image tag is defined.

#### Additional Notes ####
- Helm lint

Screenshot from 2024-04-09 15-07-19

#### After the PR is merged ####

Once the PR is merged, typically upon a new release, the necessary teams will be notified via Slack hook to perform the RKE2 Charts and RKE2 changes. Any developer working on this issue is not responsible for updating RKE2 Charts or RKE2.

### CODE REVIEW

**Comment by jiaqiluo** [comment]
Hi @Mart-Kuc, according to [k8s documentation](https://kubernetes.io/docs/concepts/containers/images/#imagepullpolicy-defaulting), the overwriting happens only when the ImagePullPolicy is not specified. So setting `ImagePullPolicy: "Always"` actually takes effect and works as expected. 

Please correct me if I misunderstand anything.

**Comment by jiaqiluo** [comment]
Hi @Mart-Kuc, yeah, it makes sense.  Just for clarification, the tag `latest` will never be used because the value will be overwritten by a specific tag from the  `versionOverrides` section.  

Instead of removing the `ImagePullPolicy` line from both files, could you convert it to a new option in the `value.yaml` and set the default value ("") , under `csiController` and `csiNode` sections? It will provide some flexibility for users who needs it.

**Review by jiaqiluo — CHANGES_REQUESTED** [review]
The changes look good to me! 
There is one last charge we need to add:  bumping the `appVersion` and `version` in the `Chart.yaml` file as we are making changes to the chart. Could you bump those two fields to `3.3.1-rancher3`?

**Comment by Mart-Kuc** [comment]
Bumped

**Comment by jiaqiluo** [comment]
Hi @brandond, could you review this PR? We need two approvals before merging it.

**Review by brandond — APPROVED** [review]

**Pull request merged**
